src/BinDeps.jl

```diff
@@ -361,26 +361,27 @@ module BinDeps
         ret
     end
 
-    @unix_only function lower(a::MakeTargets,collection)
-        cmd = `make -j8`
-        if(!isempty(a.dir))
-            cmd = `$cmd -C $(a.dir)`
-        end
-        if(!isempty(a.targets))
-            cmd = `$cmd $(a.targets)`
-        end
-        @dependent_steps ( setenv(cmd, adjust_env(a.env)), )
-    end
-    @windows_only lower(a::MakeTargets,collection) = @dependent_steps ( setenv(`make $(!isempty(a.dir)?"-C "*a.dir:"") $(a.targets)`, adjust_env(a.env)), )
+    @static if is_unix() function lower(a::MakeTargets,collection)
+            cmd = `make -j8`
+            if(!isempty(a.dir))
+                cmd = `$cmd -C $(a.dir)`
+            end
+            if(!isempty(a.targets))
+                cmd = `$cmd $(a.targets)`
+            end
+            @dependent_steps ( setenv(cmd, adjust_env(a.env)), )
+      end
+  end
+    @static is_windows() ? (lower(a::MakeTargets,collection) = @dependent_steps ( setenv(`make $(!isempty(a.dir)?"-C "*a.dir:"") $(a.targets)`, adjust_env(a.env)), )) : nothing
     lower(s::SynchronousStepCollection,collection) = (collection|=s)
 
     lower(s) = (c=SynchronousStepCollection();lower(s,c);c)
 
     #run(s::MakeTargets) = run(@make_steps (s,))
 
     function lower(s::AutotoolsDependency,collection)
-    	@windows_only prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")
-    	@unix_only prefix = s.prefix
+    	@static is_windows() ? (prefix = replace(replace(s.prefix,"\\","/"),"C:/","/c/")) : nothing
```

Comment:
It's more idiomatic to use `if end` for a single condition, the ternary requires you to go looking for a matching `:` and if what follows the `:` is just `nothing` there's no reason to use it.